### PR TITLE
connlib: JNI bridge

### PR DESCRIPTION
**Pull request by francesca64**
The biggest internal change is that all the methods on `Callbacks` (on the Rust side!) return a `Result` now, so errors from the bridge or even the client callbacks will be handled.

@roop there's nothing for you to review here, but note:

- the `bool` return values you've asked about in the past are gone now
- the route string for `onAddRoute`/`onRemoveRoute` no longer has the extra quotes (it's no longer JSON)

### CODE REVIEW

**Comment by jamilbk** [comment]
@francesca64 looks like another PR made it in and caused a merge conflict. Will merge this after resolution.

**Comment by jamilbk** [comment]
cc @conectado for approval

File: rust/connlib/libs/common/src/session.rs [hunk 1]
```diff
@@ -46,34 +50,123 @@ pub trait ControlSession<T, CB: Callbacks> {
...
+    fn on_update_resources(
+        &self,
+        resource_list: Vec<ResourceDescription>,
+    ) -> StdResult<(), Self::Error>;
     /// Called when the tunnel is disconnected.
     ///
     /// If the tunnel disconnected due to a fatal error, `error` is the error
     /// that caused the disconnect.
-    fn on_disconnect(&self, error: Option<&Error>);
+    fn on_disconnect(&self, error: Option<&Error>) -> StdResult<(), Self::Error>;
```
Comment (conectado) [inline rust/connlib/libs/common/src/session.rs:84]: I think we always ignore this error, shouldn't we just get rid of it?

**Comment by roop** [comment]
@francesca64 Okay, got it.

**Comment by jamilbk** [comment]
@francesca64 @pratikvelani I think this was just waiting on y'all for final comment / approval

**Comment by jamilbk** [comment]
Going to merge this to build upon for offsite -- can address feedback in a follow up PR.

**Pull request merged**